org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/classfmt/ClassFileReader.java

```diff
@@ -156,19 +159,24 @@ public static ClassFileReader readFromModule(
 		throws ClassFormatException, java.io.IOException {
 		return JRTUtil.getClassfile(jrt, filename, moduleName, moduleNameFilter);
 }
-public static ClassFileReader read(
-	java.util.zip.ZipFile zip,
-	String filename,
-	boolean fullyInitialize)
-	throws ClassFormatException, java.io.IOException {
+
+public static ClassFileReader read(java.util.zip.ZipFile zip, String filename, boolean fullyInitialize)
+		throws ClassFormatException, java.io.IOException {
 	java.util.zip.ZipEntry ze = zip.getEntry(filename);
-	if (ze == null)
+	if (ze == null) {
 		return null;
-	ClassFileReader classFileReader = Util.getZipEntryClassFile(zip.getName(), filename);
-	if (fullyInitialize) {
-		classFileReader.initialize();
 	}
-	return classFileReader;
+	try (BufferedInputStream stream = new BufferedInputStream(zip.getInputStream(ze))) {
```

Comment:
Since `Util.getInputStreamAsByteArray` uses `stream.readAllBytes`, the additional buffering could/should be unnecessary.

Reply:
done.

Reply:
@szarnekow can you explain why buffering is not needed with `stream.readAllBytes`

API says
> Note that this method is intended for simple cases where it is
> convenient to read all bytes into a byte array. **It is not intended for
> reading input streams with large amounts of data**

while one might argue that "large" is not the case here, buffering can still be good even if `stream.readAllBytes` is invoked? Or is Zip using some data from the archive to know the size in advance?

Reply:
readAllBytes uses internal buffers to perform bulk reads on the stream. The buffer themselves. The difference between buffered reads and direct reads is most notable if bytes are read one-by-one. That’s not the case here